dynawo/sources/Modeler/Common/DYNModelMulti.cpp

```diff
@@ -248,24 +248,66 @@ ModelMulti::initBuffers() {
 void
 ModelMulti::init(const double& t0) {
   Timer timer1("ModelMulti::init");
+
+  zSave_.resize(sizeZ(), 0.);
+  zSave_.assign(zLocal_, zLocal_ + sizeZ());
+
   // (1) initialising each sub-model
   //----------------------------------------
   for (unsigned int i = 0; i < subModels_.size(); ++i)
     subModels_[i]->initSub(t0);
 
+  // Detect if some discrete variable was modified during the initialization (e.g. subnetwork detection)
+  std::vector<double> z(sizeZ());
+  z.assign(zLocal_, zLocal_ + sizeZ());
+  vector<int> indicesDiff;
+  for (unsigned int i = 0; i < z.size(); ++i) {
+    if (doubleNotEquals(z[i], zSave_[i])) {
+      indicesDiff.push_back(i);
+    }
+  }
```

Comment:
At first glance, it is not obvious to understand why you need to introduce this additional vector for z - I finally see that you use it below as a kind of save vector.

Shouldn't it be possible to only have zSave_ and zLocal_?

It leads to something like:
...
vector<int> indicesDiff;
for (unsigned int i = 0; i < sizeZ(); ++i) {
   if (doubleNotEquals(zLocal_[i], zSave_[i])) {
        indicesDiff...
}
}
zSave_.assign(zLocal_, zLocal_ + sizeZ());

Reply:
This intermediary vector was used to be able to use the copieResultZ method to be as close as possible of evalZ